### ISSUE TITLE
Wrong slideshow order with Pillar

### ISSUE THREAD

**Issue by Ikrk** [DanielMuller/koken-plugin-photoswipe#49]
Hello Daniel,
thank you for the integration of Photoswipe to Koken - it works great for me.

However I have found an issue with the slideshow order when using Pillar.
It works good if an album has one column only. However if there is more than one column, the images are distributed in DIV elements - each column in a separate DIV - and therefore the Photoswipe init function is parsing the image elements and filling the items array by columns and not by rows. This is a problem if there is some chronological order of your photos because Koken fills the rows first.

If you want an example, in the demo of Regale here http://regale.koken.me/ the normal order of photos is first top left photo of buildings, second yellow house and so on. With Photoswipe, it would be first top left photo of building, second green mountain.

Do you have an idea how to fix that? Maybe there is some magical switch for it? :) 
Otherwise I am afraid the only solution would be to change the parseThumbnailElements function and sort the items correctly, right? 

Thank you for your answer!
Ikrk

**Comment by DanielMuller** [comment]
Same issue as described in #20, sadly nothing I can do to fix the problem.

**Comment by Ikrk** [comment]
Hi,
I have made a brute force quick&dirty solution for my custom theme using Pillar. I did not test it on any other theme, so it will be probably not very versatile (and it is not optimized in any sense :) ), but I share at least the idea if somebody is interested...

1) all div elements with pillar class that are visible (visibility:block) are found and the "size group" class is parsed from them ("size group" class parsing is probably unnecessary additional step, but it just follows the same logic, as original code)
2) number of divs with the parsed "size group" gives the number of columns currently displayed
3) all photoswipe trigger elements within parsed "size group" divs are selected 
4) knowing the number of columns and number of rows withing each column, it is iterated through all the trigger elements and the "row wise" index is transformed to "column wise" index and set as "data-pswp-uid" attribute of the trigger element.
- two important conditions - currenty max 5 columns are supported (easy to extend) and the Pillar option "balanceColumns" must be set to false as the reindexing is very simple and straightforward (not so easy to extend)
5) In the parseThumbnailElements function the "data-pswp-uid" attribute is read again and saved into the item array as "order" field
6) at the end all the items within the items array are sorted according to "order"

Check the implementation in the attached file.

Regards,
Ikrk
PS: in the attached file are some more additional customizations as for example basic video support, but I think it is pretty obvious what is what. Sorry for that - just no more time left to make it nicer for you.
[pswp.min.js.txt](https://github.com/DanielMuller/koken-plugin-photoswipe/files/1609372/pswp.min.js.txt)
